src/core/utils/types/loan-type.ts

```diff
@@ -1,7 +1,8 @@
-export interface LoanMetaDataType {
+import { ListType } from "./list-type";
+
+export interface LoanType extends ListType {
   dueDate: string | null | undefined;
   loanDate: string | null | undefined;
-  id: string;
   isRenewable: boolean;
   materialItemNumber: string | null | undefined;
```

Comment:
All the propeties that can be undefined you can instead of this:
`dueDate: string | null | undefined;`
specify it like this:
`dueDate?: string | null;`

Reply:
Changed